tests/lib/prepare.sh

```diff
@@ -256,32 +269,73 @@ prepare_classic() {
 }
 
 setup_reflash_magic() {
-        # install the stuff we need
-        distro_install_package kpartx busybox-static
-        distro_install_local_package "$GOHOME"/snapd_*.deb
-        distro_clean_package_cache
-
-        snap install "--${CORE_CHANNEL}" core
-
-        # install ubuntu-image
-        snap install --classic --edge ubuntu-image
-
-        # needs to be under /home because ubuntu-device-flash
-        # uses snap-confine and that will hide parts of the hostfs
-        IMAGE_HOME=/home/image
-        mkdir -p "$IMAGE_HOME"
-
+    # install the stuff we need
+    distro_install_package kpartx busybox-static
+    distro_install_local_package "$GOHOME"/snapd_*.deb
+    distro_clean_package_cache
+
+    # we cannot use "names.sh" here because no snaps are installed yet
+    core_name="core"
+    if is_core18_system; then
+        snap install "--${SNAPD_CHANNEL}" snapd
+        core_name="core18"
+    fi
+    snap install "--${CORE_CHANNEL}" "$core_name"
+
+    # install ubuntu-image
+    snap install --classic --edge ubuntu-image
+
+    # needs to be under /home because ubuntu-device-flash
+    # uses snap-confine and that will hide parts of the hostfs
+    IMAGE_HOME=/home/image
+    mkdir -p "$IMAGE_HOME"
+
+    # ensure that ubuntu-image is using our test-build of snapd with the
+    # test keys and not the bundled version of usr/bin/snap from the snap.
+    # Note that we can not put it into /usr/bin as '/usr' is different
+    # when the snap uses confinement.
+    cp /usr/bin/snap "$IMAGE_HOME"
+    export UBUNTU_IMAGE_SNAP_CMD="$IMAGE_HOME/snap"
+
+    if is_core18_system; then
+        # modify the snapd snap so that it has our snapd
+        UNPACKD="/tmp/snapd-snap"
```

Comment:
Can you call this UNPACK_DIR for my peace of mind please

Reply:
Sure, done